### DISCUSSION TITLE
How can I fetch "Time on Page" via the API

### DISCUSSION THREAD

**Discussion by elbaretto** [plausible/analytics#1948] [Self-Hosted Support]
Hi there

I see that this topic has been discussed in separate threads but can't seen to find an exact answer.

I have single page apps hosted and would like to get the average time a user spends on those via an API- they don't go to a different page and so the session durations are 0.

How can I achieve this?

Many thanks 🙏

**Comment by Vigasaurus** [comment]
You can't - the time on page and session duration metrics currently rely on another page view event in the same session, so only one-page viewers are often counted as both a bounce and without any session length.
